frontend/src/bundles/users/components/subscription/subscriptions.tsx

```diff
@@ -1,9 +1,76 @@
+import { Modal, RegularButton } from '~/bundles/common/components/components';
+import { DataStatus } from '~/bundles/common/enums/data-status.enum';
+import { ButtonVariant, ModalVariant } from '~/bundles/common/enums/enums';
+import {
+    useAppDispatch,
+    useAppSelector,
+    useCallback,
+    useState,
+} from '~/bundles/common/hooks/hooks';
+import {
+    subscribe,
+    unsubscribe,
+} from '~/bundles/email-subscription/store/actions';
+import { ToastType } from '~/bundles/toast/enums/show-toast-types.enum';
+import { showToast } from '~/bundles/toast/helpers/show-toast';
+
 import styles from './style.module.scss';
 import { SubscriptionItem } from './subscription-item';
+import { SubscriptionToggleItem } from './subscription-toggle-item';
 
 const Subscriptions: React.FC = () => {
```

Comment:
Don't you want to separate the business logic let's say in custom hook and keep the component as `presentational`?
As you wish

Reply:
is it ok as it is now?

Reply:
You can move everything there, and return what your component requires, for example like so, and prettify it inside custom hook

```
const {
        handleSubscribe,
        handleUnsubscribe,
        isEmailSubscriptionLoading,
        toggleItemProperties,
        isModalOpen,
        handleModalClose,
        handleUnsubscribeAndCloseModal,
    } = useEmailSubscriptions();
```